base.h

```diff
@@ -16,18 +16,17 @@ std::vector<uint16_t> b2048_encode(const bytes& in);
 bytes b2048_decode(const std::vector<uint16_t> &in);
 
 template<typename T> 
-bytes number_to_bytes(T val)
-{
-    bytes byte_array{};
-    for(int i=sizeof(val)-(8*sizeof(uint8_t)); i > sizeof(uint8_t)*8; i-=8)
-    {
-        auto byte = (val>>i)&0xFF;
-        if (0 != byte)
-        {
-            byte_array.push_back(byte);
-        }
+bytes number_to_bytes(T val) {
+  bytes byte_array{};
+
+  for(int i=sizeof(val)-(8*sizeof(uint8_t)); i > sizeof(uint8_t)*8; i-=8) {
```

Comment:
There's something weird going on here.  sizeof(x) returns its answer in bytes, but it seems you're trying to have i hold the size in bits.  For example, if val is a uint32_t, the initial i will be set to 4-8 or -4.

I think this means your loop is always being skipped, and you're only getting proper results because you encode the final byte outside the loop.

Reply:
Great catch. The intent here is to calculate the bit shifts needed for the bytes to make a byte array with leading zeroes omitted.  You're correct that this only works currently because of the final byte outside of the loop and in the use cases currently, all values are one byte wide. Upon further review, I think there's another bug in here where it will skip any zero except the last zero so I'll fix that and repush.

Reply:
I think we should eliminate number_to_bytes.  As I read the go-algorand code, threshold and version should both be single byte integers, so let's change their declaration in algorand.h to uint8_t. Then, since they are a single byte, update_address can just push_back() each one.

Reply:
Got it, I'll make this change!